### PR TITLE
Moved and revised the await operator article

### LINKED ISSUE #10741: Abridged Reference: Operators / other operators and tokens
These don't fit any group, and should have a single page for each operator.

- `?` `:` (conditional operator). Somehow, it feels out of tune with the logical operators.
- `??` (null coalescing). It also feels not in the logical operations group. Most probably, it's because I limit logical operations to those on `bool` operands.
- `+`, `+=` (string concatenation and delegate combination)
- `-`, `-=` (delegate removal)
- `::` (namespace alias qualifier)
- `nameof`
- `sizeof`
- `await`
- `new` operator
- `stackalloc` operator
- `=>` (lambda declaration)

Sub-tasks are:

- [ ] Create the new articles
- [ ] Search for existing issues on content being migrated and consolidated. This is a good opportunity to fix those.
- [ ] Redirect existing content to the new articles.
- [ ] Remove old articles.
- [ ] Update all links in the .NET Content

### LINKED ISSUE #11889: Inconsistent documentation on C# operators
The [true](../keywords/true-false-operators.md) and [false](../keywords/true-false-operators.md)  operators were added to https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/index by #11860, but they are still missing on the left-side TOC of the page as well as several others.

That page is also missing the `stackalloc` and `nameof` operators.

On the other hand, https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/operator-keywords is missing `default`,, `delegate`. `unchecked`,  `checked` are mentioned at the end, but should be on the TOC.

/cc @BillWagner @pkulikov

### PR DESCRIPTION
**Pull request by pkulikov**
Fixes #10741 
Fixes #11889 (as all operators now are under one TOC node)
Supported by dotnet/samples#1366

### CODE REVIEW

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks for all the work here @pkulikov I'm really happy with how this developed. 🥂 

I had one suggestion, and one question. Once we resolve that, it's ready to :shipit:

This is fantastic! Thanks again.

File: docs/csharp/language-reference/operators/await.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+The `async` and `await` keywords are available starting with C# 5.
+
+## await operator in the Main method
+
+Starting with C# 7.1, the [`Main` method](../../programming-guide/main-and-command-args/index.md), which is the application entry point, can be async and you can use the `await` operator in its body. In earlier C# versions, to ensure that the `Main` method waits for the completion of an asynchronous operation, you can retrieve the value of the <xref:System.Threading.Tasks.Task%601.Result?displayProperty=nameWithType> property of the <xref:System.Threading.Tasks.Task%601> instance that is returned by the corresponding async method. For asynchronous operations that don't produce a value, you can call the <xref:System.Threading.Tasks.Task.Wait%2A?displayProperty=nameWithType> method. For information about how to select the language version, see [Select the C# language version](../configure-language-version.md).
```
[the remark this answers is not in the document]
Comment (pkulikov) [reply docs/csharp/language-reference/operators/await.md]: @BillWagner I agree; I've included the updated wording in the latest commit (also saying `Task<int>`, not `Task<T>`)

**Comment by pkulikov** [comment]
@BillWagner thanks for review! I've addressed your feedback and updated the method name to follow the change made in the samples repo.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks for all the updates on this project @pkulikov 

I'll :shipit:.  This is a nice capstone to a major set of updates. Thank you.

**Comment by pkulikov** [comment]
@BillWagner thank you! A lot for all the reviews and all the discussions, for guidance and patience.

**Pull request merged**